test/model/content_test.dart

```diff
@@ -324,6 +324,58 @@ void main() {
       ]);
   });
 
+  group('parse spoilers', () {
+    testParse('with default header',
+      // ```spoiler\nhello world\n```
+      '<div class="spoiler-block"><div class="spoiler-header">\n'
+          '</div><div class="spoiler-content" aria-hidden="true">\n'
+          '<p>hello world</p>\n'
+          '</div></div>',
+      [const SpoilerNode(
+        header: [],
+        content: [ParagraphNode(links: null, nodes: [TextNode('hello world')])],
+      )],
+    );
+
+    testParse('with plain custom header',
+      // ```spoiler hello\nworld\n```
+      '<div class="spoiler-block"><div class="spoiler-header">\n'
+          '<p>hello</p>\n'
+          '</div><div class="spoiler-content" aria-hidden="true">\n'
+          '<p>world</p>\n'
+          '</div></div>',
+      [const SpoilerNode(
+        header: [ParagraphNode(links: null, nodes: [TextNode('hello')])],
+        content: [ParagraphNode(links: null, nodes: [TextNode('world')])],
+      )],
+    );
+
+    testParse('with rich header and content',
+      // ```spoiler **bold** [czo](https://chat.zulip.org/)\n*italic* [zulip](https://zulip.com/)\n```
```

Comment:
Did you find a way to provoke the header into being a list of more than one BlockContentNode? That'd be fun to also test, if so.

If not, then a test case where the header is like `1. * ## details`, showing that it's not necessarily a paragraph, would be good. (I guess more generally: it'd be good to demonstrate not-a-paragraph, either as the only BlockContentNode or not.)

Reply:
> Did you find a way to provoke the header into being a list of more than one BlockContentNode? That'd be fun to also test, if so.

No, try as I might, sadly I did not manage to do this. 😛

Reply:
No wait! I did! Here: https://chat.zulip.org/#narrow/stream/7-test-here/topic/content/near/1735802

Reply:
But that includes an image, and I don't think we can build our content widget for image previews, `MessageImage`, in a `testContentSmoke` call. That's because `MessageImage` needs `PerAccountStoreWidget.of` and `InheritedMessage.of`, and `testContentSmoke` doesn't give support for those. When we think about those dependencies, we should have this issue in mind:

- #488

Reply:
Cool.

Yeah, it's fine that `testContentSmoke` won't handle that; it's just a shortcut to enable a lot of those tests to be one line each. Instead you can write a test that's similar to what you'd get by inlining `testContentSmoke` and `prepareContentBare`, but that provides a few more enclosing widgets in the `pumpWidget` call.

(And if we find ourselves with a lot of tests like that doing similar things, we can abstract those into further helpers similar to `prepareContentBare` and/or `testContentSmoke`, or perhaps have the latter take an optional flag.)